addons/mrp/models/mrp_routing.py

```diff
@@ -105,6 +105,12 @@ def _check_no_cyclic_dependencies(self):
         if not self._check_m2m_recursion('blocked_by_operation_ids'):
             raise ValidationError(_("You cannot create cyclic dependency."))
 
+    def action_archive(self):
+        res = super().action_archive()
+        bom_lines = self.env['mrp.bom.line'].search([('operation_id', '=', self.ids)])
```

Comment:
I'm surprised `'='` works :exploding_head: but let's follow the standard

```suggestion
        bom_lines = self.env['mrp.bom.line'].search([('operation_id', 'in', self.ids)])
```